### ISSUE TITLE
CLN: remove pandas.util.testing.choice

### ISSUE THREAD

**Issue by jreback** [pydata/pandas#12386]
xref #12384 

so I think this was for numpy < 1.7 compat.

we should remove ``pandas.util.testing.choice``
https://github.com/pydata/pandas/blob/master/pandas/util/testing.py#L173

there might be some latent doc references

**Comment by jreback** [comment]
cc @carroux

**Comment by TomAugspurger** [comment]
```
pandas/tests/frame/test_query_eval.py
99:        a = tm.choice(['red', 'green'], size=10)
100:        b = tm.choice(['eggs', 'ham'], size=10)
152:        a = tm.choice(['red', 'green'], size=10)
153:        b = tm.choice(['eggs', 'ham'], size=10)
246:        a = tm.choice(['red', 'green'], size=10)
978:        a = Series(tm.choice(list('abcde'), 20))

pandas/tests/test_graphics_others.py
638:            gender = tm.choice(['male', 'female'], size=n)
712:            gender_int = tm.choice([0, 1], size=n)

pandas/tests/test_graphics.py
63:            gender = tm.choice(['Male', 'Female'], size=n)
64:            classroom = tm.choice(['A', 'B', 'C'], size=n)
3864:            gender = tm.choice(['male', 'female'], size=n)

pandas/tools/tests/test_merge.py
239:            df = DataFrame({'a': tm.choice(['m', 'f'], size=3),
241:            df2 = DataFrame({'a': tm.choice(['m', 'f'], size=10),
248:            df = DataFrame({'a': tm.choice(['m', 'f'], size=3),
251:            df2 = DataFrame({'a': tm.choice(['m', 'f'], size=10),
257:            df = DataFrame({'a': tm.choice(['m', 'f'], size=3),
259:            df2 = DataFrame({'a': tm.choice(['m', 'f'], size=10),

vb_suite/groupby.py
146:obj = tm.choice(list('ab'), size=n).astype(object)
```

**Comment by paul-reiners** [comment]
I think I have this finished.  See #12490.

**Comment by paul-reiners** [comment]
This is fixed in pull request #12505.  Tests have passed, it's all in one commit, and whatsnew has been added.
